### PR TITLE
Implements Shell Task baking

### PR DESCRIPTION
**Pull request by HavokInspiration**
I was surprised to see that it was not possible to bake a Shell Task class.
This PR implements it.

### CODE REVIEW

File: src/Shell/Task/TaskTask.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         0.1.0
```
Comment (markstory) [inline src/Shell/Task/TaskTask.php]: Needs to be updated.

File: src/Shell/Task/TestTask.php [hunk 2]
```diff
@@ -513,6 +515,10 @@ public function generateConstructor($type, $fullClassName)
             $pre = "\$this->io = \$this->getMock('Cake\Console\ConsoleIo');";
             $construct = "new {$className}(\$this->io);";
         }
+        if ($type === 'task') {
+            $pre = "\$this->io = \$this->getMock('Cake\Console\ConsoleIo');\n";
+            $construct = "\$this->getMock('{$fullClassName}', ['in', 'out', 'err', '_stop'], [\$this->io]);";
```
Comment (markstory) [inline src/Shell/Task/TestTask.php]: You shouldn't need to mock all these methods if the ConsoleIo is a mock.

**Comment by markstory** [comment]
Looking good.

File: src/Shell/Task/TestTask.php (hunk 2 above)
Comment (HavokInspiration) [inline src/Shell/Task/TestTask.php]: I used the ``*TaskTest`` example of the core to build this : all of them mock these methods when building the test subject class. See : 

- [UnloadTaskTest](https://github.com/cakephp/cakephp/blob/master/tests/TestCase/Shell/Task/UnloadTaskTest.php#L38)
- [AssetsTaskTest](https://github.com/cakephp/cakephp/blob/master/tests/TestCase/Shell/Task/AssetsTaskTest.php#L45-L48)
- and all other classes.

But I can of course remove them if they are indeed not necessary. It was just to provide consistency with what is done in the core.

**Comment by HavokInspiration** [comment]
I pushed a change with the ``@since`` modified.
Waiting to some feeding about the mock methods when baking the test file.

File: src/Shell/Task/TaskTask.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         1.2.2
```
Comment (lorenzo) [inline src/Shell/Task/TaskTask.php]: Needs update

File: tests/TestCase/Shell/Task/TaskTaskTest.php [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         0.1.0
```
Comment (antograssiot) [inline tests/TestCase/Shell/Task/TaskTaskTest.php]: you missed that one

File: src/Shell/Task/TestTask.php (hunk 2 above)
Comment (markstory) [inline src/Shell/Task/TestTask.php]: We should fix up those test cases then. Before `ConsoleIo` existed it was very necessary to mock all those methods.

Comment (HavokInspiration) [inline src/Shell/Task/TestTask.php]: Ok, I'll remove them.

**Comment by HavokInspiration** [comment]
@markstory Mock removed.
@antograssiot Changed the version number. Hopefully, there are all correct now.

**Comment by markstory** [comment]
Nice work @HavokInspiration :clap:

**Pull request merged**